src/scxml_sm_interface.cpp

```diff
@@ -46,8 +47,14 @@ static void getStateTransitionsRecursive(tinyxml2::XMLElement* state,
       throw std::runtime_error("'" + std::string(TRANSITION_ELEMENT) + "' element does not have '" +
                                std::string(EVENT_ATTRIBUTE) + "' attribute");
 
+    const char* name = transition->Attribute(TARGET_ATTRIBUTE);
+
+    std::pair<QString, QString> list = std::make_pair(QString(event), QString(name));
+
+    inherited_events.insert(list);
```

Comment:
Simplify with emplace

```suggestion
    inherited_events.emplace(QString(event), QString(transition->Attributie(TARGET_ATTRIBUTE)));
```